async/httpaf_async.ml

```diff
@@ -136,6 +145,14 @@ module Server = struct
         | `Yield ->
           (* Log.Global.printf "write_yield(%d)%!" (Fd.to_int_exn fd); *)
           Server_connection.yield_writer conn writer_thread;
+        | `Upgrade ->
+          (match upgrade_handler with
+          | None -> failwith "HTTP upgrades not supported"
+          | Some upgrade_handler ->
+            upon (upgrade_handler client_addr) (fun () ->
```

Comment:
Aren't we calling `upgrade_handler` twice for each request, once for the read loop and once for the write loop?